### DISCUSSION TITLE
Is there a formula to select tau if I have a specific uncertainty target in mind?

### DISCUSSION THREAD

**Discussion by joshua-revolutions** [OpenDebates/openskill.py#135] [Q&A]
Let's say I want players that have many games to have an uncertainty to be close to 2.5. Is there a formula I can use to know what value of tau I could use to achieve this result?

**Comment by vivekjoshy** [comment]
The way sigma is processed internally after a rating update is like so:

$$\sigma_{i} \leftarrow \sqrt{\sigma_{i}^2 + \tau^2}$$

Here is a good formula:

$$\tau = 2.5 \times \sqrt{|1 - 0.016 \times \sigma_{i}^2|}$$

In code:

```py
import math

sigma = 0
tau = 2.5 * math.sqrt(abs(1 - 0.016 * sigma**2))
processed_sigma = math.sqrt(sigma**2 + tau**2)
```

At `sigma` of `0`, it will always output `2.5` and at `sigma` of `25/2` it will output a value slightly higher than with the default `tau` of $\frac{25}{3}$.

![image](https://github.com/OpenDebates/openskill.py/assets/8206808/75f9423b-b88e-4141-a331-8549f5ddf469)

**Note:** This has not been tested on simulations or real world data, so please let us know if it works. Thanks.
